src/ToricVarieties/LineBundles/methods.jl

```diff
@@ -0,0 +1,115 @@
+############################
+# Line bundle cohomologies
+############################
+
+@doc Markdown.doc"""
+    all_cohomologies(l::ToricLineBundle)
+
+Computes the dimension of all sheaf cohomologies of the toric line bundle `l`.
+
+# Examples
+```jldoctest
+julia> dP3 = del_pezzo(3)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> all_cohomologies(ToricLineBundle(dP3, [1,2,3,4]))
+3-element Vector{fmpz}:
+ 11
+ 0
+ 0
+```
+"""
+function all_cohomologies(l::ToricLineBundle)
+    # check if we can apply cohomCalg
+    v = variety(l)
+    if !((issmooth(v) && iscomplete(v)) || (issimplicial(v) && isprojective(v)))
+        throw(ArgumentError("cohomCalg only applies to toric varieties that are either smooth, complete or simplicial, projective."))
+    end
+    
+    # extract vector of currently-known cohomology dimensions (or create it if necessary)
+    return get_attribute!(l, :all_cohomologies) do
+        
+        # Minimal example:
+        #
+        # The following lines provide an interface to cohomCalg. This is easiest explained with an example.
+        # 
+        # Example: Compute the cohomologies of the structure sheaf of the 2-dimensional projective space.
+        #
+        # Step 1: We trigger the following command so that cohomCalg computes these cohomologies for us:
+        #
+        # ./cohomCalg "vertex x1 | GLSM: ( 1 ); vertex x2 | GLSM: ( 1 ); vertex x3 | GLSM: ( 1 ); srideal [x1*x2*x3]; ambientcohom O(0 );"
+        #
+        # The notation GLSM is for generalized linear sigma model.
+        # This is a physics model which famously triggered interest in toric geometry for physics applications.
+        # The cohomCalg algorithm relies only on the grading of the Cox ring and the Stanley-Reisner ideal.
+        # <-> Those two pieces of information are (more or less) the defining data of a GLSM.
+        #
+        # Step 2: We receive the following return value from cohomCalg:
+        #
+        # "{True,{{1,0,0},{{0,1*1}}}}"
+        #
+        # The "True" tells us that the run was successful. Otherwise, we would find "False".
+        # Whenever we encounter "False", this most likely indicates wrong input data.
+        # This can be one of the following:
+        # (a) wrong variable names: cohomCalg insists on "x1, x2, ...". In particular no "[" or "]" are allowed.
```

Comment:
so all variable names must start with `x` followed by an integer? Or is also `y3` allowed?

Reply:
I have just experimented a bit more:
- `x1`, `x2`, ... works.
- `y1`, `y2`, ... works.
- `_x1`, `_x2`, ... errors.
- `x[1]`, `x[2]`, ... errors.

Hence, my comment above is not correct. I will update this shortly (working in your other suggestions as well and will push shortly). For sake of simplicity, `x1`, `x2`,... is probably easiest.